<?xml version="1.0"?>
<launch>
<arg name="robot_name" default=""/> <!-- robot_name for use NAMESPACE -->
<arg name="player_name" default="NoName"/>
<arg name="side" default="b"/> <!-- "b" = blue side, "r" = red side  -->
<arg name="ip" default="http://127.0.0.1:5000"/>

<!-- turtlebot bringup -->
<include file="$(find turtlebot3_bringup)/launch/turtlebot3_robot.launch" >
   <arg name="multi_robot_name" value="$(arg robot_name)" />
   <arg name="set_lidar_frame_id" value="$(arg robot_name)/base_scan" />
</include>

<!-- usb camera -->
<node pkg="uvc_camera" type="uvc_camera_node" name="uvc_camera_node"/>

<!-- ar reader -->
<node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher">
    <param name="use_camera_info" value="False"/>
    <remap from="/image" to="image_raw" />
    <remap from="/aruco_marker_publisher/markers" to="target_id" />
</node>

<!-- send target ID to judge server -->
<node pkg="burger_war" type="sendIdToJudge.py" name="send_id_to_judge" output="screen">
    <param name="judge_url" value="$(arg ip)" />
    <param name="player_name" value="$(arg player_name)" />
    <param name="side" value="$(arg side)" />
</node>

</launch>
